src/Symfony/Component/Form/Extension/Core/Type/CollectionType.php

```diff
@@ -81,24 +73,78 @@ public function finishView(FormView $view, FormInterface $form, array $options)
      */
     public function configureOptions(OptionsResolver $resolver)
     {
-        $entryOptionsNormalizer = function (Options $options, $value) {
-            $value['block_name'] = 'entry';
+        $entryOptionsNormalizer = function (Options $options, $entryOptions) {
+            if (is_array($entryOptions)) {
```

Comment:
btw, please add a test using such a callable to prevent regressions in its support